From b0bf234865ef9a61a0fe3657aac959e27d1c2499 Mon Sep 17 00:00:00 2001
From: Ondrej Spacek <ondrej.spacek@nxp.com>
Date: Tue, 2 Nov 2021 14:35:47 +0100
Subject: [PATCH 5/6] imagebuilder - Allow pasthrough of multiple devices.

Currently it is possible to pass single device via
"DOMU_PASSTHROUGH_DEVICE[i]".With this commit changes
variable "DOMU_PASSTHROUGH_DEVICE[i]" can contain
multiple devices to be passed through. For PFE it
is required to pass multiple nodes. This change
allows to pass all pfe slave nodes.

Example to pass PFE slave node to DOMU0:
DOMU_PASSTHROUH_DEVICE[0]="/pfe_slave@46000000 \
/pfe_slave@46000000/hif@3 /pfe_slave@46000000/ethernet@10"

Signed-off-by: Ondrej Spacek <ondrej.spacek@nxp.com>
---
 scripts/uboot-script-gen | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/scripts/uboot-script-gen b/scripts/uboot-script-gen
index 0ee2210..e2b9cd1 100755
--- a/scripts/uboot-script-gen
+++ b/scripts/uboot-script-gen
@@ -102,10 +102,17 @@ function device_tree_editing()
         then
             echo "fdt set /chosen/domU$i direct-map <0x1>" >> $UBOOT_SOURCE
         fi
-        if test "${DOMU_PASSTHROUGH_DEVICE[$i]}"
-        then
-            echo "fdt set ${DOMU_PASSTHROUGH_DEVICE[$i]} xen,passthrough" >> $UBOOT_SOURCE
-        fi
+
+        #Try to split devices, if we have more than one
+        DEV_LIST=(${DOMU_PASSTHROUGH_DEVICE[$i]})
+        DEV_CNT=${#DEV_LIST[@]}
+
+        dev=0
+        while test $dev -lt $DEV_CNT
+        do
+                echo "fdt set ${DEV_LIST[$dev]} xen,passthrough" >> $UBOOT_SOURCE
+                dev=$(( $dev + 1 ))
+        done
         i=$(( $i + 1 ))
     done
 }
-- 
2.35.1

